kernel/src/platform/mod.rs

```diff
@@ -111,6 +112,13 @@ pub trait Chip {
     /// chip and resumes the scheduler.
     fn sleep(&self);
 
+    /// Called at startup to reset any configuration from previous stages.
+    /// On some hardware it is impossible to reset the MPU after it has
+    /// been locked, in this case this function wont change those regions.
+    fn clear_mpu(&self) {
+        self.mpu().clear_mpu()
+    }
```

Comment:
Boards can do this if they need to, but I don't think we need to change this interface.

Reply:
I agree with Brad -- since this would be in `main` and is board/use case specific, it doesn't and shouldn't be in the trait.

Reply:
I have removed all the calls to this, but left it in the trait.

If it's not in the trait we will end up implementing it anyway and just making it public so that chpis/boards can access it.